receiver/signalfxreceiver/receiver_test.go

```diff
@@ -117,21 +114,66 @@ func Test_signalfxeceiver_EndToEnd(t *testing.T) {
 
 	unixSecs := int64(1574092046)
 	unixNSecs := int64(11 * time.Millisecond)
-	tsUnix := time.Unix(unixSecs, unixNSecs)
+	ts := pdata.TimestampUnixNano(time.Unix(unixSecs, unixNSecs).UnixNano())
+
 	doubleVal := 1234.5678
-	doublePt := metricstestutil.Double(tsUnix, doubleVal)
+	doublePt := pdata.NewDoubleDataPoint()
+	doublePt.InitEmpty()
+	doublePt.SetTimestamp(ts)
+	doublePt.SetValue(doubleVal)
+
 	int64Val := int64(123)
-	int64Pt := &metricspb.Point{
-		Timestamp: metricstestutil.Timestamp(tsUnix),
-		Value:     &metricspb.Point_Int64Value{Int64Value: int64Val},
+	int64Pt := pdata.NewIntDataPoint()
+	int64Pt.InitEmpty()
+	int64Pt.SetTimestamp(ts)
+	int64Pt.SetValue(int64Val)
+
+	want := pdata.NewMetrics()
+	rms := want.ResourceMetrics()
+	rms.Resize(1)
+	rm := rms.At(0)
+	rm.InitEmpty()
+
+	rm.InstrumentationLibraryMetrics().Resize(1)
+	ilm := rm.InstrumentationLibraryMetrics().At(0)
+
+	{
+		m := pdata.NewMetric()
+		m.InitEmpty()
+		m.SetName("gauge_double_with_dims")
+		m.SetDataType(pdata.MetricDataTypeDoubleGauge)
+		m.DoubleGauge().InitEmpty()
```

Comment:
(Comment unrelated to this PR, just general observation).
@bogdandrutu looks like it is a common pattern to call `SetDataType()` followed by `IntGauge().InitEmpty()` (similarly for Double, etc). Maybe we can consider having a `SetIntType()`, `SetDoubleType`, etc which does both at once. There does not seem to be a use case where `SetDataType()` is individually useful.

Reply:
Consider to file an issue to core repo.